cmd/kubeadm/app/master/clusterroles.go

```diff
@@ -0,0 +1,47 @@
+package master
+
+import (
+	"fmt"
+
+	api "k8s.io/kubernetes/pkg/api/v1"
+	rbac "k8s.io/kubernetes/pkg/apis/rbac/v1alpha1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+)
+
+// CreateBootstrapRBACClusterRole creates the necessary ClusterRole for bootstrapping
+func CreateBootstrapRBACClusterRole(clientset *clientset.Clientset) error {
+	clusterRole := rbac.ClusterRole{
+		// a role to use for setting up a proxy
+		ObjectMeta: api.ObjectMeta{Name: "system:kubelet-bootstrap"},
+		Rules: []rbac.PolicyRule{
+			rbac.NewRule("get").Groups("").Resources("nodes").RuleOrDie(),
+			rbac.NewRule("create", "watch").Groups("certificates.k8s.io").Resources("certificatesigningrequests").RuleOrDie(),
```

Comment:
@liggitt The purpose why we're doing this is not to be yet another deployment type, it's to be the lean and configurable core. I'm fine with `kubeadm:` if there absolutely is no other way, but do we have any alternatives?

This is the generic rule for a client that connects to the api in order to do the tls bootstrap.
Can we do something like `system:tls-bootstrap-token` or the like which is portable to all kinds of solutions where there might be a client that wants to connect to the API with a token in order to do the TLS bootstrap for a given node?